src/Util/File.cc

```diff
@@ -67,3 +67,88 @@ fs::path SearchPath(std::string filename) {
     }
     return fs::path();
 }
+
+CARTA::FileType GuessImageType(const std::string& path_string, bool check_content) {
+    if (check_content) {
+        // Guess file type by magic number
+        auto magic_number = GetMagicNumber(path_string);
+        switch (magic_number) {
+            case FITS_MAGIC_NUMBER:
+                return CARTA::FITS;
+            case HDF5_MAGIC_NUMBER:
+                return CARTA::HDF5;
+            case GZ_MAGIC_NUMBER:
+                fs::path gz_path(path_string);
+                return (gz_path.stem().extension().string() == ".fits") ? CARTA::FITS : CARTA::UNKNOWN;
+        }
+    } else {
+        // Guess file type by extension
+        fs::path path(path_string);
+        auto filename = path.filename().string();
+
+        if (EndsWith(filename, ".fits") || EndsWith(filename, ".FITS") || EndsWith(filename, ".fz") || EndsWith(filename, ".fits.gz")) {
```

Comment:
I would make this a case-insensitive regular expression (or three of them). What if it's `.FITS.GZ` or `.FITS.gz`, etc.?